.github/workflows/backstage-release-snapshot.yaml

```diff
@@ -0,0 +1,84 @@
+name: Publish snapshot on NPM
+
+on:
+  # When manually triggered
+  workflow_dispatch:
+
+  # When there's a new commit in main branch
+  push:
+    branches: [ "main" ]
+
+jobs:
+  publish-release-snapshot-on-npm:
+    name: Publish release on NPM
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup release name
+        run: |
+          # get the short git hash
+          git_hash=$(git rev-parse --short "$GITHUB_SHA")
+          # get timestamp
+          now=$(date +'%Y%m%d%H%M%S')
```

Comment:
The timestamp will make the action non idempotent, do we need it ?

Reply:
For 2 reasons, I want to have it:
- Better visibility: when I check the version name, it is helpful to see the time and not just the commit hash
- Support for multiple builds: something might have changed in the build process itself among 2 builds for the same commit. (not very strong opinion in this reason)

Reply:
> Better visibility: when I check the version name, it is helpful to see the time and not just the commit hash

the hash is connected to the time of the commit, build process best practices should aim for reproducible builds https://en.wikipedia.org/wiki/Reproducible_builds

> Support for multiple builds: something might have changed in the build process itself among 2 builds for the same commit. (not very strong opinion in this reason)

The build process is part of the repository so another commit is required ?

Reply:
> The build process is part of the repository so another commit is required ?

Not everything like env vars, secrets, runners you define in the repo level.

> the hash is connected to the time of the commit, build process best practices should aim for reproducible builds https://en.wikipedia.org/wiki/Reproducible_builds

Yes, but to find out how recent the snapshot I am using as a user, I need to go and check the repo, find the commit, etc. Also, the builds are still reproducible.